capstone/src/main/java/com/google/step/servlets/PageViewServlet.java

```diff
@@ -0,0 +1,132 @@
+package com.google.step.servlets;
+
+import com.google.gson.Gson;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import java.util.List;
+
+import com.google.step.clients.MetricsClient;
+import com.google.step.data.WeeklyPageView;
+import com.google.step.data.RestaurantPageViews;
+
+/**
+ * Servlet that will call the metricsClient for any doGet or doPost requests.
+ */
+@WebServlet("/page-view")
+public class PageViewServlet extends HttpServlet {
+
+    private static MetricsClient metricsClient = new MetricsClient();
+
+    /**
+     *
+     * @param request with a mix of the following parameters: <String, restaurantKey> and <int, year>.
+     *                These can be in the request or not at all, and depending on the mix a different
+     *                method from the client will be called.
+     *                String restaurantKey -> getCurrentPageViews()
+     *                int year, String restaurantKey -> getYearRestaurantPageViews()
+     *                empty -> getAllPageViews();
+     *                A SC_FORBIDDEN status will be set if none of these three specified combinations
+     *                is present.
+     * @param response Json String response that varies depending of the method called by the mix of
+     *                 parameters in the request.
+     *                 getCurrentPageViews() -> WeeklyPageView (as Json)
+     *                 getYearRestaurantPageViews() -> List<WeeklyPageView> (as Json list)
+     *                 getAllPageViews() -> List<RestaurantPageViews> (as Json list)
+     * @throws IOException can be thrown by the writer for the response.
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String restaurantKey;
+        if (request.getParameter("restaurantKey") != null) {
+            restaurantKey = request.getParameter("restaurantKey");
+        } else {
+            restaurantKey = null;
+        }
+
+        int year = 0;
+        if (request.getParameter("year") != null) {
+            year = Integer.parseInt(request.getParameter("year"));
+        }
+
+        String json;
+        if (restaurantKey == null && year == 0) {
+            // getAllPageViews()
+            List<RestaurantPageViews> restaurantPageViewsList = metricsClient.getAllPageViews();
+            json = convertRestaurantPageViewListToJsonUsingGson(restaurantPageViewsList);
```

Comment:
Is there a reason that you have this code in a helper method instead of in-lining it? It seems strange to do it this way, especially given that if the `Gson` object were a member of the servlet, it would only take one line to do this in each of the three places. You could do something like `json = gson.toJson(restaurantPageViewsList)`, etc.

Reply:
Yeah I guess it was just about custom. I changed it and made it into a variable for the class and deleted all the methods for that so now doing it in-line.